src/makeRequest.js

```diff
@@ -1,17 +1,39 @@
-import request from 'request-promise-native';
 import jwt from 'jsonwebtoken';
+import request from 'request-promise-native';
 
-export default function makeRequest(requestAttributes, { apiKey, apiSecret }) {
+export default async function makeRequest(
+  requestAttributes,
+  { apiKey, apiSecret, maxTries = 0 },
+  tryNumber = 1,
+) {
   const signed = jwt.sign({ key: apiKey }, apiSecret, { header: { kid: apiKey } });
-  return request(
-    Object.assign(
-      {
-        gzip: true,
-        auth: {
-          bearer: signed,
+  try {
+    const result = await request(
+      Object.assign(
+        {
+          gzip: true,
+          auth: {
+            bearer: signed,
+          },
         },
-      },
+        requestAttributes,
+      ),
+    );
+    return result;
+  } catch (e) {
+    if (tryNumber >= maxTries) {
+      throw e;
+    }
+    console.warn(`Failed to make request to ${requestAttributes.url}. Retrying after 2s...`);
+    await new Promise((resolve) => setTimeout(resolve, 2000));
```

Comment:
This should probably use exponential backoff and some jitter. You might consider using async-retry which has features like this built in: https://www.npmjs.com/package/async-retry